drivers/spi/spi_litespi.c

```diff
@@ -0,0 +1,173 @@
+		if (config->slave >= SPI_MAX_CS_SIZE) {
+			LOG_ERR("More slaves than supported");
+			return -ENOTSUP;
+		}
+		cs = (u8_t)(config->slave);
+	}
+
+	if (SPI_WORD_SIZE_GET(config->operation) != 8) {
+		LOG_ERR("Word size must be %d", SPI_WORD_SIZE);
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_CS_ACTIVE_HIGH) {
+		LOG_ERR("CS active high not supported");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_LOCK_ON) {
+		LOG_ERR("Lock On not supported");
+		return -ENOTSUP;
+	}
+
+	if ((config->operation & SPI_LINES_MASK) != SPI_LINES_SINGLE) {
+		LOG_ERR("Only supports single mode");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_TRANSFER_LSB) {
+		LOG_ERR("LSB first not supported");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & (SPI_MODE_CPOL | SPI_MODE_CPHA)) {
+		LOG_ERR("Only supports CPOL=CPHA=0");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_OP_MODE_SLAVE) {
+		LOG_ERR("Slave mode not supported");
+		return -ENOTSUP;
+	}
+
+	/* Set Loopback */
+	if (config->operation & SPI_MODE_LOOP) {
+		write8(SPI_ENABLE, SPI_LOOPBACK_ADDR);
+	}
+	/* Set word size */
+	*control = (u16_t) (SPI_WORD_SIZE_GET(config->operation)
+			<< POSITION_WORD_SIZE);
+	/* Write configurations */
+	write8(cs, SPI_CS_ADDR);
+	write16(*control, SPI_CONTROL_ADDR);
+
+	return 0;
+}
+
+static void spi_litespi_send(struct device *dev, u8_t frame, u16_t control)
+{
+	/* Write frame to register */
+	write8(frame, SPI_MOSI_DATA_ADDR);
+	/* Start the transfer */
+	write16(control | SPI_ENABLE, SPI_CONTROL_ADDR);
+	/* Wait until the transfer ends */
+	while (!(read8(SPI_STATUS_ADDR)))
+		;
+}
+
+static u8_t spi_litespi_recv(void)
+{
+    /* Return data inside MISO register */
+	return read8(SPI_MISO_DATA_ADDR);
+}
+
+static void spi_litespi_xfer(struct device *dev,
+		const struct spi_config *config, u16_t control)
+{
+	struct spi_context *ctx = &SPI_DATA(dev)->ctx;
+	u32_t send_len = spi_context_longest_current_buf(ctx);
+	u8_t read_data;
+
+	for (u32_t i = 0; i < send_len; i++) {
+		/* Send a frame */
+		if (i < ctx->tx_len) {
+			spi_litespi_send(dev, (u8_t) (ctx->tx_buf)[i],
+					control);
+		} else {
+			/* Send dummy bytes */
+			spi_litespi_send(dev, 0, control);
+		}
+		/* Receive a frame */
+		read_data = spi_litespi_recv();
+		if (i < ctx->rx_len) {
+			ctx->rx_buf[i] = read_data;
+		}
+	}
+	spi_context_complete(ctx, 0);
+}
+
+/* API Functions */
+
+static int spi_litespi_init(struct device *dev)
+{
+	return 0;
+}
+
+static int spi_litespi_transceive(struct device *dev,
+			  const struct spi_config *config,
+			  const struct spi_buf_set *tx_bufs,
+			  const struct spi_buf_set *rx_bufs)
+{
+	u16_t control;
+
+	spi_config(config, &control);
+	spi_context_buffers_setup(&SPI_DATA(dev)->ctx, tx_bufs, rx_bufs, 1);
+	spi_litespi_xfer(dev, config, control);
+	return 0;
+}
+
+static int spi_litespi_release(struct device *dev,
+		const struct spi_config *config)
+{
+	if (!(read8(SPI_STATUS_ADDR))) {
+		return -EBUSY;
+	}
+	return 0;
+}
```

Comment:
so your driver does not handle the lock_on attribute of spi_config?

Reply:
No it doesn't support this, in `spi_config` we verify that `SPI_LOCK_ON` is *not* set.